### PR TITLE
Improve argument rules

### PR DESCRIPTION
**Pull request by evgenyz**
Improvement of #6088.

More info pending.

### CODE REVIEW

**Comment by mildas** [comment]
b'Not building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel7/./profiles/C2S-docker.profile: documentation not complete and not a debug build\nNot building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel7/./profiles/docker-host.profile: documentation not complete and not a debug build\nNot building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel7/./profiles/http-stig.profile: documentation not complete and not a debug build\nNot building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel7/./profiles/ipa-stig.profile: documentation not complete and not a debug build\nNot building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel7/./profiles/satellite-stig.profile: documentation not complete and not a debug build\nNot building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel7/./profiles/tower-stig.profile: documentation not complete and not a debug build\n'
b'Not building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel7/./profiles/C2S-docker.profile: documentation not complete and not a debug build\nNot building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel7/./profiles/docker-host.profile: documentation not complete and not a debug build\nNot building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel7/./profiles/http-stig.profile: documentation not complete and not a debug build\nNot building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel7/./profiles/ipa-stig.profile: documentation not complete and not a debug build\nNot building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel7/./profiles/satellite-stig.profile: documentation not complete and not a debug build\nNot building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel7/./profiles/tower-stig.profile: documentation not complete and not a debug build\n'
b'Not building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel8/./profiles/ospp-mls.profile: documentation not complete and not a debug build\n'
b'Not building profile from /home/jenkins/workspace/content-test-filtering-pr/content/rhel8/./profiles/ospp-mls.profile: documentation not complete and not a debug build\n'
**Changes identified:**
Rule sssd_ldap_start_tls:
 Templatization usage changed.
Rule sssd_ldap_configure_tls_ca_dir:
 Templatization usage changed.
Rule grub2_admin_username:
 Templatization usage changed in /linux_os/guide/system/bootloader-grub2/grub2_admin_username/oval/shared.xml.
 The rule doesn't occur in any profile nor product.
Rule grub2_password:
 Templatization usage changed in /linux_os/guide/system/bootloader-grub2/grub2_password/oval/shared.xml.
Rule grub2_uefi_password:
 Templatization usage changed in /linux_os/guide/system/bootloader-grub2/grub2_uefi_password/oval/shared.xml.
Rule grub2_uefi_admin_username:
 The rule doesn't occur in any profile nor product.
 Templatization usage changed in /linux_os/guide/system/bootloader-grub2/grub2_uefi_admin_username/oval/shared.xml.
Rule checks:
 Deleted attribute from OVAL check.
 The rule doesn't occur in any profile nor product.
 Text changed in OVAL check.
 New node inserted to OVAL check.
 Node moved within OVAL check.
 Attribute value changed in OVAL check.
 Node deleted from OVAL check.
Profile moderate on rhcos4:
 Rule bls_audit_option removed from moderate profile.
 Rule coreos_audit_option added to moderate profile.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
 In Bash remediation for sssd_ldap_start_tls.
Macro oval_file_absent_criterion:
 In OVAL check for grub2_admin_username.
 In OVAL check for grub2_uefi_password.
 In OVAL check for grub2_uefi_admin_username.
 Used in template_OVAL_coreos_kernel_option template.
 In OVAL check for grub2_password.
Macro oval_file_absent:
 In OVAL check for grub2_admin_username.
 In OVAL check for grub2_uefi_password.
 In OVAL check for grub2_uefi_admin_username.
 Used in template_OVAL_coreos_kernel_option template.
 In OVAL check for grub2_password.
Macro oval_argument_in_file_criterion:
 Used in template_OVAL_zipl_bls_entries_option template.
 In OVAL check for checks.
 Used in template_OVAL_coreos_kernel_option template.
Macro oval_argument_in_file_test:
 Used in template_OVAL_zipl_bls_entries_option template.
 In OVAL check for checks.
 Used in template_OVAL_coreos_kernel_option template.
Macro oval_argument_in_file:
 Used in template_OVAL_zipl_bls_entries_option template.
 In OVAL check for checks.
 Used in template_OVAL_argument_in_file template.
 Used in template_OVAL_coreos_kernel_option template.
Others:
 Python abstract syntax tree change found in ssg/templates.py.
 Python abstract syntax tree change found in ssg/utils.py.

**Recommended tests to execute:**
 build_product fedora
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-fedora-ds.xml grub2_password
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-fedora-ds.xml grub2_uefi_password
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 build_product rhcos4
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-rhcos4-ds.xml moderate
 (cd build && cmake ../ && ctest -j4)

File: shared/templates/template_OVAL_coreos_kernel_option [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    </criteria>
+  </definition>
+
+  {{{- oval_file_absent('/boot/loader/entries/ostree-2-*.conf') }}}
+  {{{- oval_argument_in_file_test('/boot/loader/entries/ostree-1-*.conf', ARG_NAME, ARG_VALUE, 'options ') }}}
+
+  {{{- oval_argument_in_file_test('/boot/loader/entries/ostree-2-*.conf', ARG_NAME, ARG_VALUE, 'options ') }}}
+
+  {{{- oval_argument_in_file_test('/proc/cmdline', ARG_NAME, ARG_VALUE, 'BOOT_IMAGE') }}}
+</def-group>
```
Comment (JAORMX) [inline shared/templates/template_OVAL_coreos_kernel_option:30]: from a quick review, this looks good!

File: ssg/utils.py [hunk 2]
```diff
@@ -252,15 +252,28 @@ def mkdir_p(path):
...
     # See https://docs.python.org/3/library/re.html#re.sub
     # '!', '"', '%', "'", ',', '/', ':', ';', '<', '=', '>', '@', and "`" are not escaped.
-    banner_text = re.sub(r"([#$&*+-.^`|~:()])", r"\\\1", banner_text)
-    banner_text = banner_text.replace("\n", "BFLMPSVZ")
-    banner_text = banner_text.replace(" ", "[\\s\\n]+")
-    return banner_text.replace("BFLMPSVZ", "(?:[\\n]+|(?:\\\\n)+)")
+    return re.sub(r"([#$&*+-.^`|~:()])", r"\\\1", text) \
+        .replace("\n", "BFLMPSVZ") \
+        .replace(" ", "[\\s\\n]+") \
+        .replace("BFLMPSVZ", "(?:[\\n]+|(?:\\\\n)+)")
```
Comment (yuumasato) [inline ssg/utils.py:263]: I think the last 3 replaces are specific for regexes for banners.
But the first replace is generic enough.

**Review by yuumasato — COMMENTED** [review]
Isn't the `oval_argument_file` macro very similar to `line_in_file` macro?

File: ssg/utils.py (hunk 2 above)
Comment (evgenyz) [reply ssg/utils.py:263]: Hmm, hmm. Okay, maybe newlines are not really useful outside banners. But if we are creating universal filter wouldn't it be preferable to handle all cases? It's not like we are striving for tight performance in the build process itself.

**Comment by evgenyz** [comment]
> Isn't the `oval_argument_file` macro very similar to `line_in_file` macro?

Yes, it is similar. But also it is different. The `line_in_file` is tailored for a single value per line. But, if you can see a way to do the same using `line_in_file`, please give me an example.

File: ssg/utils.py (hunk 2 above)
Comment (yuumasato) [reply ssg/utils.py:263]: > But if we are creating universal filter wouldn't it be preferable to handle all cases?

I don't understand what is the purpose of this universal filter.

`banner_regexify` is not just about escaping, it also handles banner related quirks:
* [80 column folding](https://github.com/ComplianceAsCode/content/pull/6100/files#diff-6779cec91d09e974134b1e7cd05e420fR262), that is done on banners in `/etc/issue`.
* and newline [syntax differences](https://github.com/ComplianceAsCode/content/pull/6100/files#diff-6779cec91d09e974134b1e7cd05e420fR263) between `/etc/issue` and `/etc/dconf/db/gdm.d/`.

So I strongly prefer to keep `banner_regexify` separate from a general `escape_regex` function.

>  It's not like we are striving for tight performance in the build process itself.

I'm more worried about maintenance and readability, these banners can get quite complex.
See https://github.com/ComplianceAsCode/content/pull/5050 for some context.

**Review by yuumasato — COMMENTED** [review]
> Yes, it is similar. But also it is different. The `line_in_file` is tailored for a single value per line.

Thanks for clarifying the differences.

> But, if you can see a way to do the same using `line_in_file`, please give me an example.

I think `multi_value=true` can help.
https://github.com/ComplianceAsCode/content/blob/master/shared/macros-oval.jinja#L17

It is used by `oval_grub_config` macro for example.
https://github.com/ComplianceAsCode/content/blob/master/shared/macros-oval.jinja#L327

File: ssg/utils.py (hunk 2 above)
Comment (yuumasato) [reply ssg/utils.py:263]: I think that only the [first replace](https://github.com/ComplianceAsCode/content/pull/6100/commits/aa4f77c6a7c5e88c433c7fef9f82ebc2b4392604#diff-6779cec91d09e974134b1e7cd05e420fR260) is suitable for `escape_regex`.

**Comment by yuumasato** [comment]
> It is used by `oval_grub_config` macro for example.

And actually, all BLS related templates could  leverage `oval_grub_config`, :)

**Comment by mildas** [comment]
**Changes identified:**
Rule sssd_ldap_configure_tls_ca_dir:
 Templatization usage changed.
Rule sssd_ldap_start_tls:
 Templatization usage changed.
Rule grub2_uefi_admin_username:
 The rule doesn't occur in any profile nor product.
 Templatization usage changed in /linux_os/guide/system/bootloader-grub2/grub2_uefi_admin_username/oval/shared.xml.
Rule grub2_admin_username:
 The rule doesn't occur in any profile nor product.
 Templatization usage changed in /linux_os/guide/system/bootloader-grub2/grub2_admin_username/oval/shared.xml.
Rule grub2_password:
 Templatization usage changed in /linux_os/guide/system/bootloader-grub2/grub2_password/oval/shared.xml.
Rule grub2_uefi_password:
 Templatization usage changed in /linux_os/guide/system/bootloader-grub2/grub2_uefi_password/oval/shared.xml.
Rule checks:
 Attribute value changed in OVAL check.
 New node inserted to OVAL check.
 The rule doesn't occur in any profile nor product.
 Deleted attribute from OVAL check.
 Node deleted from OVAL check.
 Text changed in OVAL check.
 Node moved within OVAL check.
Macro bash_sssd_ldap_config:
 In Bash remediation for sssd_ldap_configure_tls_ca_dir.
 In Bash remediation for sssd_ldap_start_tls.
Macro oval_file_absent_criterion:
 In OVAL check for grub2_password.
 Used in template_OVAL_coreos_kernel_option template.
 In OVAL check for grub2_admin_username.
 In OVAL check for grub2_uefi_password.
 In OVAL check for grub2_uefi_admin_username.
Macro oval_file_absent:
 In OVAL check for grub2_password.
 Used in template_OVAL_coreos_kernel_option template.
 In OVAL check for grub2_admin_username.
 In OVAL check for grub2_uefi_password.
 In OVAL check for grub2_uefi_admin_username.
Macro oval_argument_in_file_criterion:
 Used in template_OVAL_coreos_kernel_option template.
 In OVAL check for checks.
 Used in template_OVAL_zipl_bls_entries_option template.
Macro oval_argument_in_file_test:
 Used in template_OVAL_coreos_kernel_option template.
 In OVAL check for checks.
 Used in template_OVAL_zipl_bls_entries_option template.
Macro oval_argument_in_file:
 Used in template_OVAL_argument_in_file template.
 Used in template_OVAL_coreos_kernel_option template.
 In OVAL check for checks.
 Used in template_OVAL_zipl_bls_entries_option template.
Others:
 Python abstract syntax tree change found in ssg/templates.py.
 Python abstract syntax tree change found in ssg/utils.py.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_configure_tls_ca_dir
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml sssd_ldap_start_tls
 (cd build && cmake ../ && ctest -j4)
 build_product fedora
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-fedora-ds.xml grub2_password
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-fedora-ds.xml grub2_uefi_password

**Comment by evgenyz** [comment]
> > Yes, it is similar. But also it is different. The `line_in_file` is tailored for a single value per line.
> 
> Thanks for clarifying the differences.
> 
> > But, if you can see a way to do the same using `line_in_file`, please give me an example.
> 
> I think `multi_value=true` can help.
> https://github.com/ComplianceAsCode/content/blob/master/shared/macros-oval.jinja#L17
> 
> It is used by `oval_grub_config` macro for example.
> https://github.com/ComplianceAsCode/content/blob/master/shared/macros-oval.jinja#L327

The only rule that use it does it like this `{{{ oval_grub_config(parameter="GRUB_CMDLINE_LINUX", value="nousb") }}}`, which is looking for value `nousb` for parameter `GRUB_CMDLINE_LINUX`. Can't imagine how it would check that `selinux=(1|yes|true)` is set.

The resulting OVAL is:
```xml
        <ind:textfilecontent54_object id="oval:ssg-obj_grub2_nousb_argument:obj:1" version="1">
          <ind:filepath>/etc/default/grub</ind:filepath>
          <ind:pattern operation="pattern match">^[ \t]*GRUB_CMDLINE_LINUX=([^#]*).*$</ind:pattern>
          <ind:instance operation="greater than or equal" datatype="int">1</ind:instance>
        </ind:textfilecontent54_object>
        <ind:textfilecontent54_state id="oval:ssg-state_grub2_nousb_argument:ste:1" version="1">
          <ind:subexpression datatype="string" operation="pattern match">^.*\bnousb\b.*$</ind:subexpression>
        </ind:textfilecontent54_state>
```

**Comment by evgenyz** [comment]
> > It is used by `oval_grub_config` macro for example.
> 
> And actually, all BLS related templates could leverage `oval_grub_config`, :)

BLS-compatible bootloader is not nesesary the GRUB, hence `/etc/default/grub` file would be irrelevant for it. Moreover, GRUB, when operating in fully BLS-compatible mode, does not use `/etc/default/grub` at all. So the assumption that some value present there would make it into the kernel is incorrect. GRUB does support BLS notation to some degree, but GRUB ≠ BLS and BLS ≠ GRUB.

**Comment by yuumasato** [comment]
> > But, if you can see a way to do the same using `line_in_file`, please give me an example.
> 
> I think `multi_value=true` can help.
> https://github.com/ComplianceAsCode/content/blob/master/shared/macros-oval.jinja#L17
> 
> It is used by `oval_grub_config` macro for example.
> https://github.com/ComplianceAsCode/content/blob/master/shared/macros-oval.jinja#L327


My point was to illustrate usage of macro `oval_check_config_file` and its argument `multi_value=true`.
Which I believe will address the use case in question.

**Comment by yuumasato** [comment]
> > > It is used by `oval_grub_config` macro for example.
> > 
> > 
> > And actually, all BLS related templates could leverage `oval_grub_config`, :)
> 
> BLS-compatible bootloader is not nesesary the GRUB, hence `/etc/default/grub` file would be irrelevant for it. Moreover, GRUB, when operating in fully BLS-compatible mode, does not use `/etc/default/grub` at all. So the assumption that some value present there would make it into the kernel is incorrect. GRUB does support BLS notation to some degree, but GRUB ≠ BLS and BLS ≠ GRUB.

My bad, you are correct in the BLS assessment.
I think I had `oval_check_config_file` in mind, but put there `oval_grub_config`.

**Comment by evgenyz** [comment]
> My point was to illustrate usage of macro `oval_check_config_file` and its argument `multi_value=true`.
> Which I believe will address the use case in question.

I got it. But still it is more tailored to check for
```
[section]
prefix parameter = value value value
```
rather that
for 
```
prefix parameter=value parameter=value parameter=value
```
While I actually was able to imagine some regexes for `oval_check_config_file` to use it for the later case, I strongly believe that it would be an abusive usage of that (already very complex) macro. Let's have a new one `oval_argument_in_file` and tailor it for the second case.

**Comment by evgenyz** [comment]
@matejak Here you are, `oval_metadata`. Is it okay to just inherit `rule_title` in these cases?

**Comment by evgenyz** [comment]
@JAORMX Can you test it, please?

**Comment by evgenyz** [comment]
Ah, dammit, conflicts

**Comment by JAORMX** [comment]
@evgenyz this worked for me!

/lgtm

**Review by JAORMX — APPROVED** [review]

File: ssg/utils.py (hunk 2 above)
Comment (evgenyz) [reply ssg/utils.py]: Refined.

**Comment by jan-cerny** [comment]
@yuumasato How do you like it? Are the ZIPL changes OK from your point of view?

File: shared/templates/template_OVAL_zipl_bls_entries_option [hunk 3]
```diff
@@ -1,43 +1,12 @@
...
-    <ind:filepath>/etc/kernel/cmdline</ind:filepath>
-    <ind:pattern operation="pattern match">^(.*)$</ind:pattern>
-    <ind:instance datatype="int" operation="greater than or equal">1</ind:instance>
-  </ind:textfilecontent54_object>
-
-  <ind:textfilecontent54_state id="state_bls_{{{ SANITIZED_ARG_NAME }}}_option"
-  version="1">
-    <ind:subexpression datatype="string" operation="pattern match">^(?:.*\s)?{{{ ESCAPED_ARG_NAME_VALUE }}}(?:\s.*)?$</ind:subexpression>
-  </ind:textfilecontent54_state>
+  {{{- oval_argument_in_file_test('/boot/loader/entries/.*.conf', ARG_NAME, ARG_VALUE, 'options ') }}}
```
Comment (yuumasato) [inline shared/templates/template_OVAL_zipl_bls_entries_option:10]: The IDs of tests for `/boot/loader/entries/.*.conf` now have 4 underscores.
`ssg-test_zipl_vsyscall_argument_vsyscall_none_argument_in_boot_loader_entries____conf`

This is not a functional problem, but not nice anyway.

File: shared/templates/template_BASH_zipl_bls_entries_option [hunk 4]
```diff
@@ -1,11 +1,11 @@
 # platform = Red Hat Enterprise Linux 8
 
 # Correct BLS option using grubby, which is a thin wrapper around BLS operations
-grubby --update-kernel=ALL --args="{{{ ARG_NAME_VALUE }}}"
+grubby --update-kernel=ALL --args="{{{ ARG_NAME }}}={{{ ARG_VALUE }}}"
 
 # Ensure new kernels and boot entries retain the boot option
 if [ ! -f /etc/kernel/cmdline ]; then
     echo "{{{ ARG_NAME_VALUE }}}" >> /etc/kernel/cmdline
```
Comment (yuumasato) [inline shared/templates/template_BASH_zipl_bls_entries_option:8]: One occurrence of `ARG_NAME_VALUE` was left over.

File: shared/templates/template_OVAL_coreos_kernel_option [hunk 5]
```diff
@@ -1,71 +1,26 @@
...
-  </unix:file_test>
-
-  <unix:file_object id="object_coreos_{{{ SANITIZED_ARG_NAME }}}_entry_2_does_not_exists"
-  version="1">
-    <unix:filepath operation="pattern match">^/boot/loader/entries/ostree-2-*\.conf</unix:filepath>
-  </unix:file_object>
-
-</def-group>
+  {{{- oval_argument_in_file_test('/proc/cmdline', ARG_NAME, ARG_VALUE, 'BOOT_IMAGE') }}}
+</def-group>
```
Comment (yuumasato) [inline shared/templates/template_OVAL_coreos_kernel_option:26]: Add a new line at the end of the file.

File: docs/manual/developer_guide.adoc [hunk 6]
```diff
@@ -1600,11 +1600,14 @@ audit_rules_usergroup_modification::
 ** *path* - path that should be part of the audit rule as a value of `-w` argument, eg. `/etc/group`.
 * Languages: Ansible, Bash, OVAL
 
-bls_bootloader_option::
-* Checks kernel command line arguments in BLS-compatible (Boot Loader Specification) boot loader configuration.
+argument_in_file::
```
Comment (yuumasato) [inline docs/manual/developer_guide.adoc:1603]: I suggest `argument_value_in_line` as a more descriptive and accurate name.

File: shared/macros-oval.jinja [hunk 7]
```diff
@@ -250,6 +251,69 @@
...
+{{%- macro oval_argument_in_file_criterion(filepath, name, value, application='') -%}}
+{{%- set name_value = name+"="+value -%}}
+  <criterion comment="Check if argument {{{ name_value }}}{{% if application %}} for {{{ application }}}{{% endif %}} is present in {{{ filepath }}}" 
+  test_ref="test_{{{ rule_id }}}_{{{ name_value|escape_id }}}_argument_in_{{{ filepath|escape_id }}}"/>
+{{%- endmacro -%}}
+
+{{#
+  Macro to define the OVAL test to check if there is a line in file with a pair of argument=value (Test definition).
+  Parameters:
+    - filepath (String): Path to the configuration file to be checked.
```
Comment (yuumasato) [inline shared/macros-oval.jinja:271]: I would add here that the operation in `filepath` is always a `pattern match`

File: shared/macros-oval.jinja [hunk 8]
```diff
@@ -250,6 +251,69 @@
...
+{{%- macro oval_argument_in_file_test(filepath, name, value, line_prefix='', line_suffix='') -%}}
+{{%- set name_value = name+"="+value -%}}
+  <ind:textfilecontent54_test id="test_{{{ rule_id }}}_{{{ name_value|escape_id }}}_argument_in_{{{ filepath|escape_id }}}"
+  comment="Check if argument {{{ name_value }}} is present{{% if line_prefix %}} in the line starting with '{{{ line_prefix }}}'{{% endif %}} in {{{ filepath }}}"
+  check="all" check_existence="all_exist" version="1">
+    <ind:object object_ref="object_{{{ rule_id }}}_{{{ name_value|escape_id }}}_argument_in_{{{ filepath|escape_id }}}" />
+    <ind:state state_ref="state_{{{ rule_id }}}_{{{ name_value|escape_id }}}_argument_in_{{{ filepath|escape_id }}}" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="object_{{{ rule_id }}}_{{{ name_value|escape_id }}}_argument_in_{{{ filepath|escape_id }}}" version="1">
+    <ind:filepath operation="pattern match">{{{ filepath }}}</ind:filepath>
```
Comment (yuumasato) [inline shared/macros-oval.jinja:286]: I recommend adding regex anchors here.
Reference: https://github.com/ComplianceAsCode/content/pull/3533

**Review by yuumasato — CHANGES_REQUESTED** [review]
> While I actually was able to imagine some regexes for `oval_check_config_file` to use it for the later case, I strongly believe that it would be an abusive usage of that (already very complex) macro. Let's have a new one `oval_argument_in_file` and tailor it for the second case.

Alright.

File: docs/manual/developer_guide.adoc (hunk 6 above)
Comment (yuumasato) [reply docs/manual/developer_guide.adoc:1603]: Or `parameter_value_in_line`.

File: shared/templates/template_OVAL_zipl_bls_entries_option (hunk 3 above)
Comment (evgenyz) [reply shared/templates/template_OVAL_zipl_bls_entries_option]: Changed `re.sub(r"[^\w]", "_", text)` to `re.sub(r"[^\w]+", "_", text)`.

File: shared/templates/template_BASH_zipl_bls_entries_option (hunk 4 above)
Comment (evgenyz) [reply shared/templates/template_BASH_zipl_bls_entries_option]: Fixed, thanks

File: shared/templates/template_OVAL_coreos_kernel_option (hunk 5 above)
Comment (evgenyz) [reply shared/templates/template_OVAL_coreos_kernel_option]: Done

File: docs/manual/developer_guide.adoc (hunk 6 above)
Comment (evgenyz) [reply docs/manual/developer_guide.adoc]: Okay, `argument_value_in_line` it is

File: shared/macros-oval.jinja (hunk 7 above)
Comment (evgenyz) [reply shared/macros-oval.jinja]: Done

File: shared/macros-oval.jinja (hunk 8 above)
Comment (evgenyz) [reply shared/macros-oval.jinja]: Done

**Review by yuumasato — APPROVED** [review]

**Pull request merged**